gradient/api_sdk/repositories/common.py

```diff
@@ -377,6 +377,84 @@ def _format_datetime(self, some_datetime):
         datetime_str = some_datetime.strftime("%Y-%m-%dT%H:%M:%SZ")
         return datetime_str
 
+class ListMetrics(GetResource):
+    OBJECT_TYPE = None
+
+    DEFAULT_INTERVAL = "30s"
+
+    @abc.abstractmethod
+    def _get_instance_by_id(self, instance_id, **kwargs):
+        pass
+
+    def _get_metrics_api_url(self, instance, protocol="https"):
+        if not instance.metrics_url:
+            raise GradientSdkError("Metrics API url not found")
+
+        metrics_api_url = concatenate_urls(protocol + "://", instance.metrics_url)
+        return metrics_api_url
+
+    def _get(self, **kwargs):
+        new_kwargs = self._get_kwargs(kwargs)
+        rv = super(ListMetrics, self)._get(**new_kwargs)
+        return rv
+
+    def _get_kwargs(self, kwargs):
+        instance_id = kwargs["id"]
+        instance = self._get_instance_by_id(instance_id)
+        started_date = self._get_start_date(instance, kwargs)
+        end = self._get_end_date(instance, kwargs)
+        interval = kwargs.get("interval") or self.DEFAULT_INTERVAL
+        metrics_api_url = self._get_metrics_api_url(instance)
+        new_kwargs = {
+            "start": started_date,
+            "interval": interval,
+            "objecttype": self.OBJECT_TYPE,
+            "handle": instance_id,
+            "metrics_api_url": "% s:8080"% metrics_api_url if 'local' in metrics_api_url else metrics_api_url,
```

Comment:
This might be too broad of a check. How about `localhost`?

Reply:
at first i assumed it would contain `localhost` but when i logged out `metrics_api_url` it said something along the lines of `http...local.paperspace...`

Reply:
oh i could check `local.paperspace` instead duh lol

Reply:
If this is specific to that domain then maybe we should be including the port in the metrics URL returned for workloads in that domain, ie. in the service layer.

Reply:
yea that would be better